learn/security/basic_security.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: Securing your project — Meilisearch documentation
+description: This tutorial will show you how to secure your Meilisearch project.
+---
+
+# Securing your project
+
+This tutorial will show you how to secure your Meilisearch project. You will see how to manage your master key and how to safely send requests to the Meilisearch API using an API key.
+
+## Creating the master key
+
+The master key is the first and most important step to secure your Meilisearch project.
+
+### Creating the master key in Meilisearch Cloud 
+
+Meilisearch Cloud automatically generates a master key for each project. This means Meilisearch Cloud projects are secure by default.
+
+You can view your master key by visiting your project overview:
+
+![An interface element named "API keys" showing three obscured keys: "Master key", "Default Search API Key", and "Default Admin API Key"](https://raw.githubusercontent.com/meilisearch/documentation/cloud-master-api-keys/assets/images/security/01-master-api-keys.png)
+
+### Creating the master key in a self-hosted instance
+
+To protect your self-hosted instance, relaunch it using the `--master-key` command-line option or the `MEILI_MASTER_KEY` environment variable:
+
+<Tabs.Container labels={['CLI', 'Environment variable']}>
+<Tabs.Content label="CLI">
+
+```sh
+./meilisearch --master-key="MASTER_KEY"
+```
+
+</Tabs.Content>
+
+<Tabs.Content label="Environment variable">
+
+UNIX:
+
+```sh
+export MEILI_MASTER_KEY="MASTER_KEY"
+./meilisearch
+```
+
+Windows:
+
+```sh
+set MEILI_MASTER_KEY="MASTER_KEY"
+./meilisearch
+```
+
+</Tabs.Content>
+</Tabs.Container>
+
+<Capsule intent="tip" title="Tools for generating a master key">
+The master key must be at least 16-bytes-long and composed of valid UTF-8 characters. Use one of the following tools to generate a secure master key:
```

Comment:
We could mention that when launching a self-hosted instance of Meilisearch without a master key, Meilisearch suggests an autogenerated one:

```
We generated a new secure master key for you (you can safely use this token):

>> --master-key qOauLmgeXv7AyYVXQwinXc0riFBJ8XRzN_ATndt6XdE <<
```

Reply:
I considered doing that, but since the idea is that users will follow each step from beginning to end, it felt weird to tell them something that only happens if they deviate from the tutorial.

Reply:
But it would make people's lives so much easier